libraries/rush-sdk/src/helpers.ts

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as path from 'path';
+import { Import, FileSystem } from '@rushstack/node-core-library';
+import type { EnvironmentVariableNames } from '@microsoft/rush-lib';
+
+export const RUSH_LIB_NAME: '@microsoft/rush-lib' = '@microsoft/rush-lib';
+export const RUSH_LIB_PATH_ENV_VAR_NAME: typeof EnvironmentVariableNames.RUSH_LIB_PATH = '_RUSH_LIB_PATH';
+
+export type RushLibModuleType = Record<string, unknown>;
+
+export class SdkContext {
+  public rushLibModule: RushLibModuleType | undefined = undefined;
+}
+
+export const sdkContext: SdkContext = new SdkContext();
+
+/**
+ * Find the rush.json location and return the path, or undefined if a rush.json can't be found.
+ *
+ * @privateRemarks
+ * Keep this in sync with `RushConfiguration.tryFindRushJsonLocation`.
+ */
+export function tryFindRushJsonLocation(startingFolder: string): string | undefined {
+  let currentFolder: string = startingFolder;
+
+  // Look upwards at parent folders until we find a folder containing rush.json
+  for (let i: number = 0; i < 10; ++i) {
```

Comment:
Arbitrary 10? Why not look at parents forever?

AFAIK it's not possible to be in an infintely deep child folder.

Reply:
This PR is simply moving the code to another file.

The code itself was ripped from `rush-lib` which does the same thing:
https://github.com/microsoft/rushstack/blob/e67788fca6b0330e1cf75facffeaf39249846efb/libraries/rush-lib/src/api/RushConfiguration.ts#L964-L977